sysinfo.sh

```diff
@@ -0,0 +1,130 @@
+#!/bin/bash
+
+###########################################################
+#
+# Author: ctrlbyte
+# Sending_Grounds Bash Scripting Challenge
+#
+# To stay in the spirit of ProLUG and Killercoda labs, 
+# script serves as a psuedo-ansible "gather-facts" clone, 
+# collecting system info from specified hosts. 
+# Assumption is that "control node" running already has 
+# setup passwordless SSH login to target nodes.
+# 
+# Requirements:
+# - Catches signals
+# - Has a help/usage statement
+# - Uses named Exit codes
+# - Has arg parsing
+# - Posts it to this repo here
+# - and the script of course works
+# 
+###########################################################
+
+# Named exit codes
+EXIT_SUCCESS=0
+EXIT_INVALID_ARGS=1
+EXIT_SIGNAL_RECEIVED=2
+
+# Print help/usage
+usage() {
+    echo "Usage: $0 [options] <hostname1> <hostname2> ..."
+    echo "  Collects system info from specified hosts"
+    echo "Options:"
+    echo "  -q          Run quick scan (default)"
+    echo "  -f          Run full scan"
+    echo "  -h          Show help/usage"
+}
+
+# Cleanup reports directory
+cleanup() {
+    echo "Cleaning up ${REPORT_DIR}"
+    rm -rf ${REPORT_DIR}
+}
+
+# Handle signals
+handle_signals() {
+    echo "Recieved signal - Exiting gracefully..."
+    cleanup
+    exit $EXIT_SIGNAL_RECEIVED
+}
+
+# Trap signals
```

Comment:
I would suggest wrapping this section in a `main(){}` because it would be easy to read through this script where you're defining functions and not realize this section will run in-line, and is not being called later, which can make it tougher to debug. 

So you'd keep the same logic especially wrt to the `trap` but just move it into main, and then call main at the very bottom with `main $@` to send all arguments passed in, into main, and from there you basically handle it the same way you're doing now with `getopts`